sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/InputStreamContent.java

```diff
@@ -100,7 +108,55 @@ public ByteBuffer toByteBuffer() {
 
     @Override
     public Flux<ByteBuffer> toFluxByteBuffer() {
-        return FluxUtil.toFluxByteBuffer(this.content.get(), STREAM_READ_SIZE);
+        if (bufferedContent != null) {
+            return Flux.fromIterable(bufferedContent).map(ByteBuffer::asReadOnlyBuffer);
+        } else {
+            return FluxUtil.toFluxByteBuffer(this.content.get(), STREAM_READ_SIZE);
+        }
+    }
+
+    @Override
+    public void writeTo(OutputStream outputStream) throws IOException {
+        InputStream inputStream = content.get();
+        if (bufferedContent != null) {
+            // InputStream has been buffered, access the buffered elements directly to reduce memory copying.
+            for (ByteBuffer bb : bufferedContent) {
+                ImplUtils.writeByteBufferToStream(bb, outputStream);
+            }
+        } else {
+            // Otherwise use a generic write to.
+            // More optimizations can be done here based on the type of InputStream but this is the initial
+            // implementation, so it has been kept simple.
+            byte[] buffer = new byte[STREAM_READ_SIZE];
```

Comment:
nit and future: maybe have an util method for it? I might be wrong, but I think we have a few places where we copy input to output stream

Reply:
Move this to calling an API that does this check.